### PR TITLE
Add blacklist

### PR DESCRIPTION
**Pull request by Akasora39**
Add blacklist backend route for inserting name into blacklist DB
Add blacklist component on frontend which renders a form for entering a name to be added in blacklist table

### CODE REVIEW

File: src/web/client/src/App.tsx [hunk 1]
```diff
@@ -9,6 +9,8 @@ import NotificationList from "./components/Notification/NotificationList";
 import Navbar from "./components/Navbar/Navbar";
 import LiveStream from "./components/LiveStream/LiveStream";
 import ProtectedRoute from "./components/ProtectedRoute/ProtectedRoute";
+import BlackList from "./components/BlackList/BlackList";
+interface Props {}
```
Comment (yliu0715) [inline src/web/client/src/App.tsx:13]: What is this?

File: src/web/client/src/App.tsx [hunk 2]
```diff
@@ -115,6 +117,15 @@ class App extends Component<Props, State> {
...
               />
+              <Route
+                exact
+                path="/blacklist"
+                render={(props) => (
+                  <BlackList
+                    {...props}
+                    isLoggedIn={this.state.isLoggedIn}
+                  />
+                )}
```
Comment (yliu0715) [inline src/web/client/src/App.tsx:128]: put this in ProtectedRoute

File: src/web/client/src/components/BlackList/BlackList.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    isLoggedIn: boolean
+}
+interface State {
+    name: string
+}
+
+/** 
+ * A component for rendering one notification
+*/
+export default class BlackList extends Component<Props, State> {
```
Comment (yliu0715) [inline src/web/client/src/components/BlackList/BlackList.tsx:14]: npm run lint

File: src/web/client/src/App.tsx (hunk 1 above)
Comment (chc5) [inline src/web/client/src/App.tsx:13]: sup

File: src/db/cass/create_db.py [hunk 4]
```diff
@@ -58,6 +58,9 @@
 # Table for storing Notifications
 cluster_services.create_table_schema('CREATE TABLE IF NOT EXISTS notif (date timestamp, type text, email text, name text, frame_id text, stream_id text, PRIMARY KEY(date))')
 
+# Table for Blacklist
+cluster_services.create_table_schema('CREATE TABLE IF NOT EXISTS blacklist (name text), PRIMARY KEY(name))')
```
Comment (chc5) [inline src/db/cass/create_db.py:62]: Screenshot from 2020-05-06 22-37-12
I got this error from this

Comment (Akasora39) [reply src/db/cass/create_db.py]: Had an extra parenthesis there. Shouldn't have this issue now.

File: src/web/server/app/routes/blacklist.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+const app = express();
+
+app.get(
+  '/insertBlacklist',
+  async (req: express.Request, res: express.Response) => {
+    blacklistController.insertBlacklist(req, res);
+  }
+);
+
```
Comment (chc5) [inline src/web/server/app/routes/blacklist.ts:12]: Export app out

File: src/web/client/src/components/BlackList/BlackList.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+  /**
+   * Submits the name to be added to blacklist table in DB
+   * @params event - An event that records submission of a form element
+   */
+  onSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    try {
+      event.preventDefault();
+      const res = await fetch("/blacklist/insertBlacklist", {
+        method: "POST",
```
Comment (chc5) [inline src/web/client/src/components/BlackList/BlackList.tsx:41]: You asked a POST but in the server it was a GET

File: src/web/server/app/routes/blacklist.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import express from 'express';
+import * as blacklistController from '../controllers/blacklist';
+
+const app = express();
+
+app.get(
```
Comment (chc5) [inline src/web/server/app/routes/blacklist.ts:6]: You asked for POST in the client, but this is a GET

**Comment by chc5** [comment]
Screenshot from 2020-05-07 14-24-13

**Pull request merged**